test/e2e/gameserver_test.go

```diff
@@ -90,6 +90,38 @@ func TestSDKSetLabel(t *testing.T) {
 	assert.NotEmpty(t, gs.ObjectMeta.Labels["stable.agones.dev/sdk-timestamp"])
 }
 
+func TestHealthCheckDisable(t *testing.T) {
+	t.Parallel()
+	gs := defaultGameServer()
+	gs.Spec.Health = v1alpha1.Health{
+		Disabled:            true,
+		FailureThreshold:    1,
+		InitialDelaySeconds: 1,
+		PeriodSeconds:       1,
+	}
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+	defer framework.AgonesClient.StableV1alpha1().GameServers(defaultNs).Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
+
+	_, err = e2eframework.PingGameServer("UNHEALTHY", fmt.Sprintf("%s:%d", readyGs.Status.Address,
```

Comment:
nit: curious why doesn't PingGameServer just take the gameserver object ? We seem to be always doing this Sprintf which is just ugly

Reply:
Yeah, I wondered the same thing when i was writing this! That would be a good refactor. I'll make a separate PR.